File: app/src/main/java/yapp/dev_diary/Calendar/Activity/BaseActivity.java
package yapp.dev_diary.Calendar.Activity;

import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import yapp.dev_diary.Calendar.interfaces.IInitializer;

/**
 * Created by 1100160 on 2017. 3. 17..
 */

public class BaseActivity extends AppCompatActivity implements IInitializer {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    protected void initialize() {
        initData();
        initView();
        initControl();
    }

    @Override
    public void initData() {

    }

    @Override
    public void initView() {

    }

    @Override
    public void initControl() {

    }
}

File: app/src/main/java/yapp/dev_diary/baseActivity.kt
package yapp.dev_diary

import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

abstract class
baseActivity<T: ViewDataBinding> : AppCompatActivity()  {

    lateinit var viewDataBinding : T

    abstract val layoutResourceId : Int

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        viewDataBinding = DataBindingUtil.setContentView(this, layoutResourceId)
    }
}